resources/definitions/fdmprinter.def.json

```diff
@@ -1379,6 +1347,31 @@
                     "limit_to_extruder": "wall_0_extruder_nr",
                     "settable_per_mesh": true
                 },
+                "min_feature_size":
+                {
+                    "label": "Minimum Feature Size",
+                    "description": "Minimum thickness of thin features. Model features that are thinner than this value will not be printed, while features thicker than the Minimum Feature Size will be widened to the Minimum Wall Line Width.",
+                    "unit": "mm",
+                    "default_value": 0,
+                    "minimum_value": "0",
+                    "maximum_value": "wall_line_width_0",
+                    "type": "float",
+                    "limit_to_extruder": "wall_0_extruder_nr",
+                    "enabled": "fill_outline_gaps"
+                },
+                "min_bead_width":
+                {
+                    "label": "Minimum Wall Line Width",
+                    "description": "Width of the wall that will replace thin features (according to the Minimum Feature Size) of the model. If the Minimum Wall Line Width is thinner than the thickness of the feature, the wall will become as thick as the feature itself.",
+                    "unit": "mm",
+                    "value": "wall_line_width_0",
+                    "minimum_value": "0.5 * machine_nozzle_size",
```

Comment:
Ah, I initially misread this, but Cura sets hard limits (`minimum_value` and `maximum_value`) only for physical or technical impossibilities. Examples of this are negative speeds (physical impossibility), layer height more than the printer height (technical constraint), or more than 20 gradual infill steps (technical impossibility due to integer overflows).
This limit here is just because we think it won't print well, so the user should be allowed to choose minimum wall line widths of lower than half a nozzle size, but should be warned when they do so.

Reply:
From our discussion on Friday, the takeaway I intended to give is that I thought that half a line width was a good default for this setting. The minimum value should probably be 0.001mm, or maybe 0.01mm if rounding errors become a technical limitation.

Reply:
Let's make it into a minimum_value_warning then. It makes more sense than to strictly restrict the users.